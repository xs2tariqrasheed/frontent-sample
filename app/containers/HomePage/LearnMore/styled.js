import styled from 'styled-components';
const clouds = 'https://www.blockletegames.com/images/clouds-repeatable.png';
const flag = 'https://www.blockletegames.com/images/flag.png';

export const Container = styled.div`
  background-image: url(${clouds});
  background-repeat: repeat-x;
  background-position: 0 1.5rem;
  background-color: #2bb673;
  border-bottom: 87px solid #37e893;
  position: relative;
  &:after {
    content: '';
    display: block;
    background-image: url(${flag});
    background-size: contain;
    width: 81px;
    height: 146px;
    position: absolute;
    bottom: -25px;
    right: 20%;
    @media all and (max-width: 640px) {
      right: 0;
    }
    @media all and (max-width: 375px) {
      display: none;
    }
  }
`;

export const GolfBallSection = styled.section`
  padding-left: 5rem;
  position: relative;
  &:after {
    content: '';
    display: block;
    background-image: url(${props => props.bg});
    background-size: contain;
    background-repeat: no-repeat;
    object-fit: contain;
    width: 56px;
    height: 57px;
    position: absolute;
    top: 0;
    left: 0;
  }
`;
