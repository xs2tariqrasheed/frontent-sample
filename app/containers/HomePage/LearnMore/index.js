/*
 * LearnMore
 *
 * This is the LearnMore block on the front page.
 *
 */
import React from 'react';
const golfBallOne = 'https://www.blockletegames.com/images/golf-ball-1.png';
const golfBallTwo = 'https://www.blockletegames.com/images/golf-ball-2.png';
const golfBallThree = 'https://www.blockletegames.com/images/golf-ball-3.png';
const golfBallFour = 'https://www.blockletegames.com/images/golf-ball-4.png';
import { Link } from 'react-router-dom';
import { Container, GolfBallSection } from './styled';

export default function LearnMore() {
  return (
    <Container className="bg-green-400 text-white">
      <article className="container mx-auto px-4 py-20">
        <div className="sm:grid grid-rows-2 grid-flow-col gap-4">
          <section className="row-span-2 md:row-span-2">
            <h1 className="callout-underline text-4xl font-extrabold uppercase">
              How To Play and Earn Cash
            </h1>
            <p className="w-3/5 mt-5">
              Play golf while you earn! See why people across the globe are
              testing their digital golf skills for a chance at cash.
              <br />
              <Link
                className="triangle-link inline-block mt-6 self-end text-white font-extrabold no-underline uppercase text-sm"
                to="/faq"
              >
                How To Play Blocklete™ Golf
              </Link>
            </p>
          </section>

          <GolfBallSection
            className="mt-5 mb-10 md:mt-0 md:row-span-1"
            bg={golfBallOne}
          >
            <h2>Pick Your Golfer!</h2>
            <p>
              Pick the golfer that best suits your game play style. Lesser
              skilled golfers can grind their way up, while higher skilled
              golfers are competition ready!
            </p>
          </GolfBallSection>

          <GolfBallSection className="md:row-span-1 mb-10" bg={golfBallTwo}>
            <h2>Earn from Training!</h2>
            <p>
              List your golfer as a trainer so you can earn cash. Buy a trainer
              to level up your golfer’s skills to win in competitions.
            </p>
          </GolfBallSection>

          <GolfBallSection className="md:row-span-1" bg={golfBallThree}>
            <h2>Compete for Medals and Cash!</h2>
            <p>
              Winning against your competitor’s will earn you medals that level
              up your golfer or earn you cash prizes. Over $20K in cash prizes
              were won last season!
            </p>
          </GolfBallSection>

          <GolfBallSection className="md:row-span-1" bg={golfBallFour}>
            <h2>Sell Leveled Up Golfers!</h2>
            <p>
              Once you have leveled your golfer up, list it on the marketplace
              and earn cash back! Rinse, Repeat, Earn!
            </p>
          </GolfBallSection>
        </div>
      </article>
    </Container>
  );
}
