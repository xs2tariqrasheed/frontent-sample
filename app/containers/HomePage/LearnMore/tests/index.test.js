import React from 'react';
import { render } from 'react-testing-library';
import { BrowserRouter } from 'react-router-dom';
import LearnMore from '../index';

describe('<LearnMore />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <BrowserRouter>
        <LearnMore />
      </BrowserRouter>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
