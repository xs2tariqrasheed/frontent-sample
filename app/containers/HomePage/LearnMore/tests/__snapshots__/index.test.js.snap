// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LearnMore /> should render and match the snapshot 1`] = `
<div
  class="bg-green-400 text-white styled__Container-sc-1yjhil0-0 fqBHn"
>
  <article
    class="container mx-auto px-4 py-20"
  >
    <div
      class="sm:grid grid-rows-2 grid-flow-col gap-4"
    >
      <section
        class="row-span-2 md:row-span-2"
      >
        <h1
          class="callout-underline text-4xl font-extrabold uppercase"
        >
          How To Play and Earn Cash
        </h1>
        <p
          class="w-3/5 mt-5"
        >
          Play golf while you earn! See why people across the globe are testing their digital golf skills for a chance at cash.
          <br />
          <a
            class="triangle-link inline-block mt-6 self-end text-white font-extrabold no-underline uppercase text-sm"
            href="/faq"
          >
            How To Play Blocklete™ Golf
          </a>
        </p>
      </section>
      <section
        class="mt-5 mb-10 md:mt-0 md:row-span-1 styled__GolfBallSection-sc-1yjhil0-1 bTKkEI"
      >
        <h2>
          Pick Your Golfer!
        </h2>
        <p>
          Pick the golfer that best suits your game play style. Lesser skilled golfers can grind their way up, while higher skilled golfers are competition ready!
        </p>
      </section>
      <section
        class="md:row-span-1 mb-10 styled__GolfBallSection-sc-1yjhil0-1 bTKkEI"
      >
        <h2>
          Earn from Training!
        </h2>
        <p>
          List your golfer as a trainer so you can earn cash. Buy a trainer to level up your golfer’s skills to win in competitions.
        </p>
      </section>
      <section
        class="md:row-span-1 styled__GolfBallSection-sc-1yjhil0-1 bTKkEI"
      >
        <h2>
          Compete for Medals and Cash!
        </h2>
        <p>
          Winning against your competitor’s will earn you medals that level up your golfer or earn you cash prizes. Over $20K in cash prizes were won last season!
        </p>
      </section>
      <section
        class="md:row-span-1 styled__GolfBallSection-sc-1yjhil0-1 bTKkEI"
      >
        <h2>
          Sell Leveled Up Golfers!
        </h2>
        <p>
          Once you have leveled your golfer up, list it on the marketplace and earn cash back! Rinse, Repeat, Earn!
        </p>
      </section>
    </div>
  </article>
</div>
`;
